geosolutions-it/csi-sira
# SIRA Access Manager advanced (issue #314)

#314 

 - a.k.a. the "_Rules Engine_"
 - added the capability to define complex conditions to be dynamically evaluated at runtime with [`Spring Expression Language (SpEL)`](http://docs.spring.io/spring/docs/3.1.4.RELEASE/spring-framework-reference/html/expressions.html) at version `3.1.4`
 - having authenticated user informations (i.e.: `InfoPersona`) as implicit evaluation context for dynamic evaluations
 - dynamic evaluated complex expressions are available at various levels of `Rule` configuration, namely:
   - rule access mode
   - rule filter
   - rule hidden attributes
     - this configuration element has been greatly enhanced, offering the possibility to express a tree of complex, dynamically evaluated conditions to select which attributes to hide
     - the expression tree is modeled after [`XSLT` choose-when-otherwise construct](http://www.w3schools.com/xml/xsl_choose.asp)
     - the possibility to express a simple list of attributes to hide as been preserved, for backward compatibility
 - with unit-tests
   - the test-coverage is quite high, hovering around 75%. _Nevertheless, a much-needed thoroughly test session may obviously still pinpoint bugs and/or "stinky" implementations, with respect to the profound changes made_ (this is for you, @diegosanmartino :wink:)
 - other minor changes
   - ensure that authentication principal is not forced to its string representation
   - ensure that all authentication credentials got erased as soon as possible after a successful authentication attempt

_that's all folks!_ :raised_hands:
